src/settings/TargetProfileForm.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Row, Col, TextField, Button } from '@folio/stripes/components';
+import stripesFinalForm from '@folio/stripes/final-form';
+
+class TargetProfileForm extends React.Component {
+  static propTypes = {
+    pristine: PropTypes.bool.isRequired,
+    submitting: PropTypes.bool.isRequired,
+    initialValues: PropTypes.object,
+    handleSubmit: PropTypes.func.isRequired,
+  };
+
+  render() {
+    const { pristine, submitting, initialValues, handleSubmit } = this.props;
+
+    return (
+      <form
+        id="form-loan-rules"
+        data-test-circulation-rules-form
+        onSubmit={handleSubmit}
+      >
+        <Row end="xs">
+          <Col xs={3}>
+            <FormattedMessage id="ui-inventory.name">
+              {placeholder => (
+                <TextField
+                  aria-label={placeholder}
```

Comment:
I think I took the standard approach here of finding some other form control that works and passes ESLint, and copying into my code. I long ago lost track of which combination of `label`, `title` and `aria-label` is currently considered acceptable, and it brings me no comfort to hear that this is changing yet again right now. If you will just tell me what the currently correct thing to do here is, I will gladly change to that (so long as ESLint consents).

Reply:
I believe `aria-label` _has_ to be a string, which `<FormattedMessage>` is _not_, thus this render-props approach which does provide a string. Since May 2020** it's also safe to use the `injectIntl` HOC (which will provide `props.intl.formatMessage()`) or the `useIntl` hook (which provides `intl.formatMessage()`) to get at the string version. I readily admit this has been fussy and a bit error-prone; hopefully the usage patterns around `react-intl` have finally settled down a bit. 

** when we upgraded to `v4`; currently we're on `v5`. Until then, we were on `v2` and its `injectIntl` impl used legacy context, which is why we pushed toward the render-props usage, even though that's super clunky.

Reply:
So it looks like there are two things going on here:
1. @doytch thinks ui-inventory should not be using the render-props approach because that was deprecated (for reasons unknown) in the ERM project; but @zburke thinks it's a perfectly cromulent way to get the string version of the translation.
2. The code uses this string as the value of the `aria-label` and `placeholder` props of the `<TextField>`, but @doytch thinks they should be used for different props — I am not sure which.

I don't have a position on either of these things (though I would love to know for my own satisfaction what the right answers are). But since this is not part of the code that I touched in the present PR, I propose to ignore both problems for now, treating them as SEPs. Changes to the control labelling should be handed in a separate PR if it needs addressing.

Reply:
I didn’t express an opinion on render-props. My point is that you don’t have a ‘label’ on this field and only use a ‘placeholder’. Placeholders disappear when the user starts writing and Folio/Filip have advocated not using placeholders at all because of their issues.